File: 이코테/큰 수의 법칙.cpp
#include <iostream>
#include <stdio.h>
#include <vector>
#include <algorithm>
using namespace std;
int main()
{
    int n,m,k, first, second;
    int tmp;
    int result=0, count =0;
    vector<int> x;
    
    cin>>n>>m>>k;
    for(int i=0;i<n;i++)
    {
        cin>>tmp;
        x.push_back(tmp);
    }
    
    sort(x.begin(),x.end());
    
    first=x[n-1];
    second=x[n-2];
    
    count = m/(k+1)*k; //가장 큰 수가 더해지는 횟수
    count += m%(k+1); // 가장 큰 수가 추가로 더해지는 횟수
    
    result += (count)*first; // 가장 큰 수 더하기
    result += (m-count)*second; //두 번째로 큰 수 더하기
    
    cout << '\n'<< result;
    return 0;
}

File: 이코테/큰 수의 법칙.py
#n,m,k를 공백으로 구분하여 입력받기
n,m,k = map(int, input().split())

#n개의 수(배열)를 공백으로 구분하여 입력받기
data = list(map(int, input().split()))

data.sort() #배열 정렬

first = data[n-1] #가장 큰 수
second = data[n-2] #두 번쨰로 큰 수

result = 0

while True:
    for i in range(k): #가장 큰 수를 k번 더하기
        if m == 0: #m이 0이면 break
            break
        result += first
        m -= 1
    if m==0: #m이 0이면 바깥 반복문도 탈출
        break
    result += second
    m -= 1

print(result) #결과 출력
